### ISSUE TITLE
Dealing with CPU/GPU in QPFunctionFn's backward

### ISSUE THREAD

**Issue by oumayb** [Simple-Robotics/proxsuite#296]
When the arguments of the `QPFunctionFn`'s `forward` are on GPU, `rhs[:dim] = dl_dzhat[I]`, `rhs[dim : dim + neq] = dl_dlams[I]` and `rhs[dim + neq :] = dl_dnus[I]` fail because `rhs` is a numpy array on CPU, and `dl_dzhat `, `dl_dlams `  and `dl_dnus ` are torch.tensors located on GPU. 
I'm working on a PR to fix this. 
https://github.com/Simple-Robotics/proxsuite/blob/15cde910722fd74e08ce7aafadd9a1e7755cc148/bindings/python/proxsuite/torch/qplayer.py#L196

PR: https://github.com/Simple-Robotics/proxsuite/pull/297
